src/Components/Color/ColorCard.jsx

```diff
@@ -0,0 +1,72 @@
+import "./Color.css";
+import "./Buttons.css";
+import { useState } from "react";
+import ColorForm from "./ColorForm";
+
+export default function Color({ color, onDelete, onChange }) {
+  const [isEditing, setIsEditing] = useState(false);
+  const [confirmDelete, setConfirmDelete] = useState(null);
+
+  function handleEdit() {
+    setIsEditing(true);
+  }
+
+  function handleChange(updatedColor) {
+    onChange(color.id, updatedColor);
+    setIsEditing(false);
+  }
+
+  function handleCancelEdit() {
+    setIsEditing(false);
+  }
+  function handleDelete() {
+    setConfirmDelete(true);
+  }
+  function handleDeleteCancel() {
+    setConfirmDelete(null);
+  }
+  function handleDeleteConfirm() {
+    onDelete(color.id);
+    setConfirmDelete(false);
+  }
+
+  return (
+    <div
+      className="color-card"
+      style={{
+        background: color.hex,
+        color: color.contrastText,
+      }}
+    >
+      {isEditing ? (
+        <>
+          <h3 className="color-card-headline">{color.hex}</h3>
+          <h4>{color.role}</h4>
+          <p>contrast: {color.contrastText}</p>
+          <ColorForm
+            addColor={handleChange}
+            defaultValues={color}
+            isEditing={true}
+          />
+          <button onClick={handleCancelEdit}>CANCEL</button>
+        </>
+      ) : (
+        <>
+          <h3 className="color-card-highlight">{color.hex}</h3>
+          <h4>{color.role}</h4>
+          <p>contrast: {color.contrastText}</p>
+          <button onClick={handleEdit}>Edit</button>
+          {confirmDelete ? (
+            <div className="delete-confirmation">
+              <p className="color-card-headline">Really delete?</p>
+              <button onClick={handleDeleteConfirm}>DELETE</button>
+              <button onClick={handleDeleteCancel}>CANCEL</button>
+            </div>
+          ) : (
+            <button onClick={handleDelete}>Delete</button>
+          )}
+        </>
+      )}
```

Comment:
the h3, h4 and p tag will be displayed no matter what the edit state. You can put it outside the ternary operator to avoid duplication:
```suggestion
          <h3 className="color-card-highlight">{color.hex}</h3>
          <h4>{color.role}</h4>
          <p>contrast: {color.contrastText}</p>
      {isEditing ? (
        <>
          <ColorForm
            addColor={handleChange}
            defaultValues={color}
            isEditing={true}
          />
          <button onClick={handleCancelEdit}>CANCEL</button>
        </>
      ) : (
        <>
          <button onClick={handleEdit}>Edit</button>
          {confirmDelete ? (
            <div className="delete-confirmation">
              <p className="color-card-headline">Really delete?</p>
              <button onClick={handleDeleteConfirm}>DELETE</button>
              <button onClick={handleDeleteCancel}>CANCEL</button>
            </div>
          ) : (
            <button onClick={handleDelete}>Delete</button>
          )}
        </>
      )}
```

Reply:
https://github.com/Jennifer-Hinrichsen/recap-project-4/pull/12/commits/ed37302e7a38c333a145dbc8e1e055f57a93d923